transform/livereloadinject/livereloadinject.go

```diff
@@ -72,23 +57,9 @@ func New(baseURL url.URL) transform.Transformer {
 		c := make([]byte, len(b))
 		copy(c, b)
 
-		if idx == -1 {
-			idx = len(b)
-			match = tag{warnRequired: true}
-		}
-
 		script := []byte(fmt.Sprintf(`<script src="%s" data-no-instant defer></script>`, html.EscapeString(src)))
 
-		i := idx
-		if match.appendScript {
-			i += bytes.Index(b[i:], []byte(">")) + 1
-		}
-
-		if match.warnRequired {
-			script = append(script, []byte(warnScript)...)
-		}
-
-		c = append(c[:i], append(script, c[i:]...)...)
```

Comment:
And again here, doesn't `append(script, c[i:]...)` cause another unnecessary reallocation? Wouldn't it be better to do:

    c := make([]byte, 0, len(b)+len(script))
    c = append(c, b[:idx]...)
    c = append(c, script...)
    c = append(c, b[idx:]...)

Reply:
From my quick glance, I don't think you're right here, but it would be trivial to write a quick benchmark.